modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/SleepTimer.kt

```diff
@@ -16,10 +18,19 @@ import javax.inject.Singleton
 import au.com.shiftyjelly.pocketcasts.localization.R as LR
 
 @Singleton
-class SleepTimer @Inject constructor(@ApplicationContext private val context: Context) {
+class SleepTimer @Inject constructor(
+    private val analyticsTracker: AnalyticsTrackerWrapper,
+    @ApplicationContext private val context: Context,
+) {
 
     companion object {
         private var sleepTimeMs: Long? = null
+        private var lastSleepAfterTimeInMinutes: Int = 0
+        private var lastTimeHasFinishedInMillis: Long = 0
+        private var lastEpisodeUuidAutomaticEnded: String? = null
```

Comment:
Can we move this properties out of `companion object`? For now it would be only a cosmetic change but it isn't good to keep state in global static objects. And `SleepTimer` is marked with `@Singleton` ensuring that Dagger will provide only single instance of it.

Reply:
I agree. [721153b](https://github.com/Automattic/pocket-casts-android/pull/2048/commits/721153bdc98495e648717e05f994416fcd0b44d3)